src/briefcase/platforms/macOS/utils.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+
+import concurrent
+import email
+import hashlib
+import subprocess
+from pathlib import Path
+
+from briefcase.exceptions import BriefcaseCommandError
+
+
+def sha256_file_digest(path: Path) -> str:
+    """Compute a sha256 checksum digest of a file.
+
+    :param path: The file to digest.
+    :returns: A sha256 hex digest for the file.
+    """
+    with path.open("rb") as f:
+        file_hash = hashlib.sha256()
+        while chunk := f.read(8192):
+            file_hash.update(chunk)
+
+    return file_hash.hexdigest()
+
+
+class AppPackagesMergeMixin:
+    # A mixin containing the utilities to merge independent platform-specific app_packages folders
+    # into a single "fat" app_packages folder. This is currently only used by macOS, but it *could*
+    # be required on iOS if they ever re-introduce multiple on-device architectures.
+
+    def find_binary_packages(
+        self,
+        install_path: Path,
+        universal_suffix: str = None,
+    ) -> list[tuple[str, str]]:
+        """Find the packages that have been installed that have binary components.
+
+        :param install_path: The path into which packages have been installed.
+        :param universal_suffix: The tag suffix that indicates a universal wheel.
+        :returns: A list of (package name, version) tuples, describing the packages that
+            are in the install path that are non-universal and non-pure.
+        """
+        binary_packages = []
+        for distinfo in install_path.glob("**/*.dist-info"):
+            # Read the WHEEL file in the dist-info folder.
+            # Use this to determine if the wheel is "pure", and the tag
+            # for the wheel.
+            with (distinfo / "WHEEL").open("r", encoding="utf-8") as f:
+                wheel_data = email.message_from_string(f.read())
+                is_purelib = wheel_data["Root-Is-Purelib"] == "true"
```

Comment:
The wheel spec doesn't explicitly say that this key is required.

```suggestion
                is_purelib = wheel_data.get("Root-Is-Purelib", "false") == "true"
```